src/FSharpPlus/Extensions/Task.fs

```diff
@@ -91,6 +91,49 @@ module Task =
                         ) |> ignore) |> ignore
             tcs.Task
 
+    /// <summary>Creates a task workflow from three workflows 'x', 'y' and z, mapping its results with 'f'.</summary>
+    /// <remarks>Workflows are run in sequence.</remarks>
+    /// <param name="f">The mapping function.</param>
+    /// <param name="x">First task workflow.</param>
+    /// <param name="y">Second task workflow.</param>
+    /// <param name="z">Third task workflow.</param>
+    let map3 (f : 'T -> 'U -> 'V -> 'W) (x : Task<'T>) (y : Task<'U>) (z: Task<'V>) : Task<'W> =
+        if x.Status = TaskStatus.RanToCompletion && y.Status = TaskStatus.RanToCompletion && z.Status = TaskStatus.RanToCompletion then
+            try Task.FromResult (f x.Result y.Result z.Result)
+            with e ->
+                let tcs = TaskCompletionSource<'W> ()
+                tcs.SetException e
+                tcs.Task
+        else
+            let tcs = TaskCompletionSource<'W> ()
+            match x.Status, y.Status, z.Status with
+            | TaskStatus.Canceled, _                  , _                   -> tcs.SetCanceled ()
+            | TaskStatus.Faulted , _                  , _                   -> tcs.SetException x.Exception.InnerExceptions
+            | _                  , TaskStatus.Canceled, _                   -> tcs.SetCanceled ()
+            | _                  , TaskStatus.Faulted , _                   -> tcs.SetException y.Exception.InnerExceptions
+            | _                  , _                  , TaskStatus.Canceled -> tcs.SetCanceled ()
+            | _                  , _                  , TaskStatus.Faulted  -> tcs.SetException z.Exception.InnerExceptions
+            | _                  , _                  , _                   ->
+                x.ContinueWith (
+                    function
+                    | Canceled    -> tcs.SetCanceled ()
+                    | Faulted e   -> tcs.SetException e.InnerExceptions
+                    | Completed r ->
+                        y.ContinueWith (
+                                function
```

Comment:
We should use 4 spaces indentation.
I think you did it like this because I did the same mistake in `map2`, feel free to correct it there as well.